app/src/ui/history/commit-list.tsx

```diff
@@ -39,42 +39,42 @@ interface ICommitListProps {
   readonly localCommitSHAs: ReadonlyArray<string>
 
   /** The message to display inside the list when no results are displayed */
-  readonly emptyListMessage: JSX.Element | string
+  readonly emptyListMessage?: JSX.Element | string
 
   /** Callback which fires when a commit has been selected in the list */
-  readonly onCommitsSelected: (
+  readonly onCommitsSelected?: (
     commits: ReadonlyArray<Commit>,
     isContiguous: boolean
   ) => void
 
   /** Callback that fires when a scroll event has occurred */
-  readonly onScroll: (start: number, end: number) => void
+  readonly onScroll?: (start: number, end: number) => void
 
   /** Callback to fire to undo a given commit in the current repository */
-  readonly onUndoCommit: ((commit: Commit) => void) | undefined
+  readonly onUndoCommit?: ((commit: Commit) => void) | undefined
 
   /** Callback to fire to reset to a given commit in the current repository */
-  readonly onResetToCommit: (commit: Commit) => void
+  readonly onResetToCommit?: (commit: Commit) => void
 
   /** Callback to fire to revert a given commit in the current repository */
-  readonly onRevertCommit: ((commit: Commit) => void) | undefined
+  readonly onRevertCommit?: ((commit: Commit) => void) | undefined
```

Comment:
I think we can remove `| undefined`, no?
```suggestion
  readonly onRevertCommit?: (commit: Commit) => void
```